cf-testd/cf-testd.c

```diff
@@ -632,8 +633,32 @@ static void *CFTestD_ServeReport(void *config_arg)
             Log(LOG_LEVEL_ERR, "Error reading report file '%s'", report_file);
             exit(EXIT_FAILURE);
         }
-        config->report     = StringWriterClose(contents);
-        config->report_len = SafeStringLength(config->report);
+
+        size_t report_data_len = StringWriterLength(contents);
+        char *report_data = StringWriterClose(contents);
+
+        Seq *report = SeqNew(64, NULL);
+        size_t report_len = 0;
+
+        StringRef ts_ref = StringGetToken(report_data, report_data_len, 0, "\n");
+        char *ts = (char *) ts_ref.data;
+        *(ts + ts_ref.len) = '\0';
+        SeqAppend(report, ts);
+
+        /* start right after the newline after the timestamp header */
+        char *position = ts + ts_ref.len + 1;
+        char *row;
+        size_t row_len;
+        while (CFTestD_GetRow(position, &row, &row_len))
```

Comment:
I would prefer if you call it "report lines" instead of rows, to attempt some consistency: https://github.com/cfengine/nova/blob/master/REPORTING.md

(I know this has not been good in the past, and in CSV it's called rows).